cluster-autoscaler/cloudprovider/test/test_cloud_provider.go

```diff
@@ -120,6 +121,27 @@ func (tcp *TestCloudProvider) GetAvailableGPUTypes() map[string]struct{} {
 	}
 }
 
+// GetNodeAccelerator returns the label and resource name for the accelerator added to node.
+func (tcp *TestCloudProvider) GetNodeAccelerator(node *apiv1.Node) cloudprovider.Accelerator {
+	gpuLabel := tcp.GPULabel()
+	if gpu.NodeHasGpu(gpuLabel, node) {
+		return cloudprovider.Accelerator{Label: gpuLabel, Type: node.Labels[gpuLabel], ResourceName: gpu.ResourceNvidiaGPU}
+	}
+	return cloudprovider.Accelerator{Label: "", Type: "", ResourceName: ""}
+}
+
+// GetAvailableAccelerators returns all available accelerators the cloud provider supports.
+func (tcp *TestCloudProvider) GetAvailableAccelerators() []cloudprovider.Accelerator {
+	availableAccelerators := []cloudprovider.Accelerator{}
+	gpuLabel := tcp.GPULabel()
+	availableGPUTypes := tcp.GetAvailableGPUTypes()
+	for acceleratorType := range availableGPUTypes {
+		accelerator := cloudprovider.Accelerator{Label: gpuLabel, Type: acceleratorType, ResourceName: gpu.ResourceNvidiaGPU}
+		availableAccelerators = append(availableAccelerators, accelerator)
+	}
+	return availableAccelerators
+}
+
```

Comment:
I don't like how there is a semi-identical implementation copy for every cloud provider. That way 
1. We are duplicating a lot of code - this commit has 2k lines, it could easily be a fraction of that.
2. I some implementations we have 2 different accelerator info source of truth - accelerator functions you added should always use `GpuLabel()` and `GetAvailableGPUTypes()` methods, otherwise we risk that somebody will change only one of them and the behaviour will diverge.

I think you should add the implementation to a separate method which takes the `CloudProvider` as an argument. Then you can use it in every CloudProvider implementation, this should be much cleaner.

Reply:
* Created two functions in "cloud_provider.go" called `GetNodeAcceleratorFromCloudProvider()` and `GetAvailableAcceleratorsFromCloudProvider()`. Each cloud provider now calls these functions, and these functions work using `GPULabel()` and `GetAvailableGPUTypes()`.
* However, in order not to create a cyclic dependency between `gpu` and `cloudprovider` packages, each cloud provider has to check whether a node has an accelerator (by calling `gpu.NodeHasGpu()`) before calling  `GetNodeAcceleratorFromCloudProvider()`. This is the best solution I could come up with.

Reply:
Hmm, what if you moved these two functions to `gpu` package instead? This way you could use the `NodeHasGpu` function without cyclic imports. I also like this option because:
1. `cloudprovider.go` is mostly interfaces used in a lot of places in the code. Adding implementations there is not that great - if we would want to change the implementation in the future in some way that require imports, then we could potentially create a whole bunch of non trivial cyclic imports.
2. Both `GetNodeAcceleratorFromCloudProvider()` and `GetAvailableAcceleratorsFromCloudProvider()` implementations are GPU specific, so adding them in a `gpu.go` seems more intuitive. I would also change their names to  `GetNodeGPUFromCloudProvider()` and `GetAvailableGPUFromCloudProvider()` for the same reason.

Reply:
Moved those two functions to the `gpu` package and renamed as suggested.